src/txdb.cpp

```diff
@@ -147,7 +147,7 @@ size_t CCoinsViewDB::EstimateSize() const
     return db.EstimateSize(DB_COIN, (char)(DB_COIN+1));
 }
 
-CBlockTreeDB::CBlockTreeDB(size_t nCacheSize, bool fMemory, bool fWipe) : CDBWrapper(GetDataDir() / "blocks" / "index", nCacheSize, fMemory, fWipe) {
+CBlockTreeDB::CBlockTreeDB(size_t nCacheSize, bool fMemory, bool fWipe) : CDBWrapper(gArgs.IsArgSet("-blocksdir") ? GetDataDir() / "blocks" / "index" : GetBlocksDir() / "index", nCacheSize, fMemory, fWipe) {
```

Comment:
What is this conditional for -- why not always just `GetDataDir() / "blocks" / "index"` ?

Reply:
The `-blocksdir` parameter lets you put blocks in a completely different directory, e.g. on a slow cheap magnetic harddrive, keeping the rest on your datadir on SSD. `blocks/index` also needs to be on a fast drive. Ideally tough I think datadir/blocks/index should move to datadir/index/blocks

Reply:
Yes, I just mean that no matter whether the `gArgs.IsArgSet("-blocksdir") ?` conditional here is True / False, either way it results in using "DATADIR/blocks/index". So the conditional seems like just extra code ... though this makes me suspect I'm missing something.

Reply:
`gArgs.IsArgSet("-blocksdir") ? GetDataDir() / "blocks" / "index" : GetBlocksDir() / "index"`

If launched without `-blocksdir`:
```
false -> GetBlocksDir() / "index" -> ~/.bitcoin/blocks/index
```

If launched with with `-blocksdir=/mnt/magnetic/blocks`:
```
false -> GetDataDir() / "blocks" / "index" ->   ~/.bitcoin/blocks/index
```

Note that `GetBlocksDir()` internally also checks the `-blocksdir` argument, and create a directory if needed.

https://github.com/bitcoin/bitcoin/blob/a1926362ecb3c354ae338ef7d7020daf78f980c9/src/util.cpp#L607-L634

Reply:
Ah sorry, I meant to say either way it uses `GetBlocksDir()/"blocks"/"index"` but wrote DATADIR. Anyway though as you say, the directory check&create does make it slightly different.

Reply:
> If launched with with `-blocksdir=/mnt/magnetic/blocks`:
> 
> ```
> true -> GetDataDir() / "blocks" / "index" ->   /mnt/magnetic/blocks/index
> ```

This is incorrect right @Sjors?

Reply:
I believe it's like this: `-blocksdir=/mnt/magnetic` will put blocks in `/mnt/magnetic/blocks`. The block index will remain at `~/.bitcoin/blocks/index` and `~/.bitcoin/blocks` will otherwise be empty.